lib/witsec/anonymizer.rb

```diff
@@ -1,103 +1,84 @@
-                total = 0
-                row_batches.each_with_index do |batch, index|
-                  print "Anonymizing up to row #{total + batch.size} of #{rows.size}\r"
-                  total += batch.size
-                  values = batch.map do |row|
-                    "(#{row.map { |value| ActiveRecord::Base.connection.quote(value) }.join(", ")})"
-                  end.join(", ")
-
-                  output_connection.execute(
-                    "INSERT INTO #{table_name} (#{columns.join(", ")}) VALUES #{values}"
-                  )
-                end
-              end
-
-              output_connection_pool.release_connection
-            else
-              puts "Skipping #{table_name}"
-              next
+        clear_output_database
+
+        input_database.tables.each do |table_name|
+          next unless schema.anonymizes?(table_name)
+
+          # A performance improvement could probably be found here, if we just passed along included tables (as in tables, where no rows are anonymized) without querying etc.
+          record_rows = input_database[table_name].all
+          columns = record_rows&.first&.keys
+          rows = record_rows.map(&:values)
+          puts "Anonymizing #{table_name} (#{rows.size} rows)"
+
+          anonymized_rows = Witsec::Alias.new(table_name, columns:, schema:).anonymize(rows)
+
+          row_batches = anonymized_rows.in_groups_of(BATCH_SIZE, false)
+          total = 0
+          row_batches.each_with_index do |batch, index|
+            print "Anonymizing up to row #{total + batch.size} of #{rows.size}\r"
+            total += batch.size
+
+            disable_output_referential_integrity do
+              output_database[table_name].import(columns, batch)
             end
           end
         end
       end
       puts "Anonymized all in #{time.real} seconds"
     end
 
+    private
+
     def clear_output_database
       puts "Clearing output database"
 
-      ActiveRecord::Base.logger.silence do
-        connection = output_connection_pool.lease_connection
-
-        connection.disable_referential_integrity do
-          connection.tables.each do |table_name|
-            connection.execute("TRUNCATE TABLE #{table_name} CASCADE")
-          end
-        end
-
-        output_connection_pool.release_connection
+      output_database.tables.each do |table_name|
+        output_database.run("TRUNCATE TABLE #{table_name} CASCADE")
       end
     end
 
-    private
-
     def check_input_and_output_are_different
-      return if Rails.env.test?
+      return if defined?(Rails) && Rails.env.test?
 
-      if input_connection_pool.lease_connection.current_database == output_connection_pool.lease_connection.current_database
-        raise Witsec::InputAndOutputDatabasesAreTheSame, "You've probably forgotten to setup the output database. It must be named anonymized."
+      if Witsec.config.output == Witsec.config.input
```

Comment:
Ved du at man kan sammenligne `dry-configurable` objects?
Så vidt jeg ved har vi ikke en test for dette, så tænker det måske er lidt relevant lige at tjekke inden vi sletter flere databaser ;)